### ISSUE TITLE
Error on OS X 10.11

### ISSUE THREAD

**Issue by cache0928** [rofl0r/proxychains-ng#74]
Is proxychains-ng with some errors  on OS X 10.11?

**Comment by rofl0r** [comment]
it shouldn't. which error do you experience and which version of proxychains-ng are you using ?

**Comment by Willian-Zhang** [comment]
Error on OS X 10.11 +1

**Comment by rofl0r** [comment]
which error ?

**Comment by Willian-Zhang** [comment]
generally it doesn't works.

- version
``` bash
$ brew info proxychains-ng
proxychains-ng: stable 4.10, HEAD
Hook preloader
https://sourceforge.net/projects/proxychains-ng/
/usr/local/Cellar/proxychains-ng/4.8.1 (8 files, 100K)
  Built from source
/usr/local/Cellar/proxychains-ng/4.10 (8 files, 88K) *
  Built from source
From: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/proxychains-ng.rb
==> Options
--universal
	Build a universal binary
--HEAD
	Install HEAD version
```

**Comment by Willian-Zhang** [comment]
# testcase
## Config
``` bash
strict_chain
proxy_dns
remote_dns_subnet 224
tcp_read_time_out 15000
tcp_connect_time_out 8000
localnet 127.0.0.0/255.0.0.0

[ProxyList]
socks5 127.0.0.1 30039
```
## Test
```
$ proxychains4 curl http://ifconfig.co/
[proxychains] config file found: /Users/Willian/.proxychains/proxychains.conf
[proxychains] preloading /usr/local/Cellar/proxychains-ng/4.10/lib/libproxychains4.dylib
{{an ip that has direct accessed http://ifconfig.co/ }}
```

**Comment by rofl0r** [comment]
why does your proxychains not print version information ? (it should)

**Comment by Willian-Zhang** [comment]
like `[proxychains] DLL init: proxychains-ng 4.10` ?
on my non-el-capitan OS X, it does and it works

**Comment by rofl0r** [comment]
if that is so, someone needs to research which changes were introduced in 10.11 that breaks LD_PRELOAD and how to fix it.

**Comment by rofl0r** [comment]
possibly it's the guilt of this new "feature": https://en.wikipedia.org/wiki/System_Integrity_Protection

**Comment by Willian-Zhang** [comment]
any idea how to fix it?

**Comment by rofl0r** [comment]
you could try to disable the system integrity protection (as described in the wikipedia article) temporarily to see if it makes proxychains work. if so, we can try to seek a permanent workaround.

**Comment by cache0928** [comment]
<img width="682" alt="2015-08-22 5 48 58" src="https://cloud.githubusercontent.com/assets/7350995/9423439/2f4ca1bc-48f6-11e5-894c-0f67c7f38b75.png">
I can't install proxychains-ng by homebrew on OS X 10.11

**Comment by rofl0r** [comment]
@cache0928 it seems your toolchain is not properly set up, so it cannot find your system headers. i suspect you need to install "XCode" and configure homebrew to use it.

**Comment by Willian-Zhang** [comment]
> you could try to disable the system integrity protection (as described in the wikipedia article) temporarily to see if it makes proxychains work. if so, we can try to seek a permanent workaround.

yes, disabling it would led to success in working of proxychains

**Comment by Willian-Zhang** [comment]
### Disabling SIP
Run `csrutil disable` in Recovery mode

**Comment by rofl0r** [comment]
@Willian-Zhang thanks for testing. now we can look if there's a way to disable the LD_PRELOAD "protection" "feature" for handpicked applications, in our case proxychains-ng.

**Comment by Willian-Zhang** [comment]
thank u and good luck with it

**Comment by Willian-Zhang** [comment]
@rofl0r any luck with this problem?

**Comment by rofl0r** [comment]
i haven't found any documentation about the LD_PRELOAD specific protection so far. we need to wait until more information is available (probably after the official release).

**Comment by doublelduob** [comment]
xcode-select -s /Applications/Xcode-beta.app/Contents/Developer

then 

brew install roxychains-ng

**Comment by cache0928** [comment]
@doublelduob It helps when I install proxychains-ng by homebrew. But proxychains-ng also can't work if I don't disable the SIP

**Comment by Willian-Zhang** [comment]
I should probably separate LD_PRELOAD to a different issue
